jax/_src/dlpack.py

```diff
@@ -82,16 +83,108 @@ def to_dlpack(x: Array, take_ownership: bool = False,
       x.addressable_data(0), stream=stream
   )  # type: ignore
 
+def _place_array(_arr, device, dlpack_device, copy):
+  if device and dlpack_device != device:
+    if copy == False:
```

Comment:
I think `if copy is not None and not copy` are the correct semantics here.

Reply:
For `copy in {None, True, False}` they should be equivalent semantics right? I'm fine with the change, just wondering if there is an edge case I'm overlooking 😅

Reply:
I don't think we should assume that people will write code with only those three singletons, especially for truthy values.

Reply:
A specific example of where the original code would differ from the code in my suggestion would be something like `copy = []`.

Reply:
A pythonic alternative to
```python
if copy == False: ...
```
is
```python
if not copy: ...
```
that will cover @jakevdp example as well.

IIUC, the default value to copy is `False` (when `copy=None` is used), so the above remains correct when no `copy` argument is specifed.

Reply:
But if `copy = None`, then `not copy` returns True. `None` has different semantics than `False` in this context, which is why I checked for it in my suggested code.

Reply:
Right, so in this particular case use:
```python
if copy is not None and not copy:
  raise ValueError(...)
else:
  return device_put(_arr, device)
```

Reply:
As you suggested above :)

Reply:
> I don't think we should assume that people will write code with only those three singletons, especially for truthy values.

Gotcha, this was the operating assumption I was missing. I was unsure how strict we wanted to be with such values. Thanks for the clarification!